packages/component-library/components/Spacing/Spacing.elm

```diff
@@ -0,0 +1,249 @@
+        Padding n ->
+            config |> setSpacing "padding" n
+
+        PaddingTop n ->
+            config |> setSpacing "padding-top" n
+
+        PaddingRight n ->
+            config |> setSpacing "padding-right" n
+
+        PaddingBottom n ->
+            config |> setSpacing "padding-bottom" n
+
+        PaddingLeft n ->
+            config |> setSpacing "padding-left" n
+
+        PaddingX n ->
+            config |> setSpacing "padding-x" n
+
+        PaddingY n ->
+            config |> setSpacing "padding-y" n
+
+        Margin n ->
+            config |> setSpacing "margin" n
+
+        MarginTop n ->
+            config |> setSpacing "margin-top" n
+
+        MarginRight n ->
+            config |> setSpacing "margin-right" n
+
+        MarginBottom n ->
+            config |> setSpacing "margin-bottom" n
+
+        MarginLeft n ->
+            config |> setSpacing "margin-left" n
+
+        MarginX n ->
+            config |> setSpacing "margin-x" n
+
+        MarginY n ->
+            config |> setSpacing "margin-y" n
+
+
+type Property
+    = Padding GridFraction
+    | PaddingTop GridFraction
+    | PaddingRight GridFraction
+    | PaddingBottom GridFraction
+    | PaddingLeft GridFraction
+    | PaddingX GridFraction
+    | PaddingY GridFraction
+    | Margin GridFraction
+    | MarginTop GridFraction
+    | MarginRight GridFraction
+    | MarginBottom GridFraction
+    | MarginLeft GridFraction
+    | MarginX GridFraction
+    | MarginY GridFraction
+
+
+setSpacing : String -> GridFraction -> Config -> Config
+setSpacing cssProperty value config =
+    config |> Dict.insert cssProperty value
+
+
+
+-- view
+
+
+inlineStyles : Config -> List (Attribute msg)
+inlineStyles config =
+    config
+        |> Dict.toList
+        |> sortSpacingProperties
+        |> List.map
+            (\( property, value ) ->
+                let
+                    cssValue =
+                        gridFractionToCssValue value
+                in
+                if not (isCssProperty property) then
+                    case property of
+                        "margin-x" ->
+                            [ style "margin-left" cssValue
+                            , style "margin-right" cssValue
+                            ]
+
+                        "margin-y" ->
+                            [ style "margin-top" cssValue
+                            , style "margin-bottom" cssValue
+                            ]
+
+                        "padding-x" ->
+                            [ style "padding-left" cssValue
+                            , style "padding-right" cssValue
+                            ]
+
+                        "padding-y" ->
+                            [ style "padding-top" cssValue
+                            , style "padding-bottom" cssValue
+                            ]
+
+                        _ ->
+                            []
+
+                else
+                    [ style property cssValue ]
```

Comment:
Are we deferring the decision to introduce https://github.com/rtfeldman/elm-css here?

Reply:
Yeah, I certainly thought about, but this is the problem, you'd need to either:
A: have one elm-css instantation per fox instance (and make elm-css an implementation detail that consumers don't know about). Maybe it's fine having one `<style>` element per box, even if there are hundreds of boxes on the page, I'm really not sure, and it would take time to asses the performance of that.
B: get all consumers to buy into using elm-css, and having to do the elm-css boilerplate, just to use the box component which is mostly meant to speed you up. If decide to go all in on elm-css, it makes sense to do it this way.

Reply:
Is there any elm-css code in production at Culture Amp?

Reply:
Not that I'm aware of.
Those are good points though 👍